accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -673,6 +627,18 @@ The use of `typedef` for something which is not a type may be too confusing. Ano
 extension MyWidgetList<T extends Widget> = prefix.MyList<T>;
 ```
 
+### Avoiding Scope Conflicts
+
+We allow importing two extensions with the same name into the same scope, and both of them *work*. They cannot be referenced by name, but they can be implicitly invoked. This breaks the current invariant that we cannot have two declarations with the same name in the same scope.
+
+To retain this invariant, we can define a "fresh extension alias" for each extension declaration. It is a second *fresh name* for the extension (a name not occurring anywhere else in the program), which will be a fresh *private* name if the extension's name is private (or omitted, in which case we only need the one fresh name), and a fresh non-private name if the extension's name is non-private. Any implicit extension invocation uses the *fresh extension alias* name to refer to the extension (so an `expr.foo()` which resolves to a method on the extension  `Ext` is equivalent to `$fresh$Ext(expr).foo()`). 
+
+A show/hide modifier on an import or export for the name *N* must also show or hide any fresh alias which is an alias for something named *N*. So, for each fresh name, we need to be able to determine the non-fresh name it corresponds to.
```

Comment:
I don't think it makes sense to "show" a fresh name, except if you're thinking in a whole program context.  There is no reasonable scheme whereby I can modularly compile two independent modules and arrange for the two sets of fresh names chosen to be globally fresh.  The "freshness" is local to the compilation unit.  So there's no point in exporting the "fresh" names - they're not guaranteed not to collide anyway.

Reply:
I'm sure a compiler can find a way (perhaps encode the library URI in the name).

I think I'll just drop this idea again. It was intended as a different view on how scoping works, making extensions accessible if their fresh name was in scope, whether or not their declared name was, but it's dangerous to thread a new feature through the needleholes cut for a different feature.

Reply:
I don't believe that the compiler can find a way (other than probabilistically, by choosing a large random number, in which case it is impossible for another module to *predict* what name is chosen).  If something external to the compiler (like the programmer) is required to associate a globally unique key with every compilation unit (are library URIs globally unique?  I think they aren't since you can use paths, right?) then yes, you can bootstrap off of that, but the compiler *cannot* produce a globally unique name in a predictable way without an external source of predictable uniqueness.